packages/backend/src/open_payments/payment/incoming/service.ts

```diff
@@ -220,3 +264,33 @@ async function getAccountIncomingPaymentsPage(
     accountId: accountId
   })
 }
+
+async function updateIncomingPayment(
+  deps: ServiceDependencies,
+  { id, state }: UpdateIncomingPaymentOptions
+): Promise<IncomingPayment | IncomingPaymentError> {
+  return deps.knex.transaction(async (trx) => {
+    const payment = await IncomingPayment.query(trx)
+      .findById(id)
+      .forUpdate()
+      .withGraphFetched('account.asset')
+    if (!payment) return IncomingPaymentError.UnknownPayment
+    const update: PartialModelObject<IncomingPayment> = {}
+    if (state === IncomingPaymentState.Completed) {
+      switch (payment.state) {
+        case IncomingPaymentState.Pending:
+        case IncomingPaymentState.Processing:
+          update.state = state
+          update.active = false
+          break
+        default:
+          return IncomingPaymentError.InvalidState
```

Comment:
This could be something like [`WrongState`](https://github.com/interledger/rafiki/blob/9ee418a32a22debe98ac4528071e069a65d0c0d6/packages/backend/src/open_payments/payment/outgoing/service.ts#L207) to distinguish from an invalid `updateIncomingPayment` state parameter